uni-stark/tests/mul_air.rs

```diff
@@ -3,28 +3,52 @@ use std::marker::PhantomData;
 use itertools::Itertools;
 use p3_air::{Air, AirBuilder, BaseAir};
 use p3_baby_bear::{BabyBear, DiffusionMatrixBabybear};
-use p3_challenger::DuplexChallenger;
+use p3_challenger::{DuplexChallenger, HashChallenger, SerializingChallenger32};
+use p3_circle::{Cfft, CirclePcs};
 use p3_commit::testing::TrivialPcs;
 use p3_commit::ExtensionMmcs;
 use p3_dft::Radix2DitParallel;
 use p3_field::extension::BinomialExtensionField;
 use p3_field::{AbstractField, Field};
 use p3_fri::{FriConfig, TwoAdicFriPcs};
+use p3_keccak::Keccak256Hash;
 use p3_matrix::dense::RowMajorMatrix;
 use p3_matrix::MatrixRowSlices;
 use p3_merkle_tree::FieldMerkleTreeMmcs;
+use p3_mersenne_31::Mersenne31;
 use p3_poseidon2::Poseidon2;
-use p3_symmetric::{PaddingFreeSponge, TruncatedPermutation};
+use p3_symmetric::{
+    CompressionFunctionFromHasher, PaddingFreeSponge, SerializingHasher32, TruncatedPermutation,
+};
 use p3_uni_stark::{prove, verify, StarkConfig, StarkGenericConfig, Val, VerificationError};
 use rand::distributions::{Distribution, Standard};
 use rand::{thread_rng, Rng};
 
 /// How many `a * b = c` operations to do per row in the AIR.
-const REPETITIONS: usize = 10;
+const REPETITIONS: usize = 20;
 const TRACE_WIDTH: usize = REPETITIONS * 3;
 
+/*
+In its basic form, asserts a^(self.degree-1) * b = c
+(so that the total constraint degree is self.degree)
+
+
+If `uses_transition_constraints`, checks that on transition rows, the first a = row number
+*/
 pub struct MulAir {
```

Comment:
I feel like it'd be a bit nicer to have a few AIRs where each does one simple operation, like factorial, or repeated exponentiation by some constant. We could break it up later though.